### PR TITLE
Allow query through router when load moving average extension

### PR DESCRIPTION
**Pull request by bananaaggle**
A work around to make moving average query through router.
 Hi, @FrankChen021. As I mentioned in #11262, add an empty implementation of QuerySegmentWalker in CliRouter.java can make moving average query through router(just load this extension in broker and router). We use it on production for months. I don't think it very suitable for make same change in community version, because this change only for a contribution extension and it is wired to add an empty implementation. I hope it can help you. If you think my way is suitable, I will change my code following your advice. And if you have other good idea, you can open another PR. 

This PR has:
- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: services/src/main/java/org/apache/druid/cli/FakeQuerySegmentWalker.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package org.apache.druid.cli;
+
+import org.apache.druid.query.Query;
+import org.apache.druid.query.QueryRunner;
+import org.apache.druid.query.QuerySegmentWalker;
+import org.apache.druid.query.SegmentDescriptor;
+import org.joda.time.Interval;
+
+public class FakeQuerySegmentWalker implements QuerySegmentWalker
```
Comment (nishantmonu51) [inline services/src/main/java/org/apache/druid/cli/FakeQuerySegmentWalker.java:9]: nit: rename to NoopQuerySegmentWalker to be consistent with other druid naming conventions.

File: services/src/main/java/org/apache/druid/cli/CliRouter.java [hunk 2]
```diff
@@ -91,6 +92,8 @@ public CliRouter()
           JsonConfigProvider.bind(binder, "druid.router.avatica.balancer", AvaticaConnectionBalancer.class);
           JsonConfigProvider.bind(binder, "druid.router.managementProxy", ManagementProxyConfig.class);
 
+          binder.bind(QuerySegmentWalker.class).to(FakeQuerySegmentWalker.class).in(LazySingleton.class);
```
Comment (nishantmonu51) [inline services/src/main/java/org/apache/druid/cli/CliRouter.java:95]: I would also add a comment mentioning that Router is just used to route the query to appropriate broker, so it is bound to a Noop walker.

**Review by nishantmonu51 — COMMENTED** [review]
this seems like a reasonable fix for the moving average query. left some minor comments.

**Comment by bananaaggle** [comment]
@nishantmonu51 Class changed. I don't think its' package is proper, should I move this  NoopQuerySegmentWalker to other package? And should I create unit test for this change?

**Comment by FrankChen021** [comment]
IMO, there should be an IT case to cover this change. But since there's no IT integration for community contributed extensions, currently we don't have to do so. 

And for the changes in this PR, any javadoc to describe why `NoopQuerySegmentWalker` should be binded in router is useful.

**Comment by bananaaggle** [comment]
> why `NoopQuerySegmentWalker` should b

OK I will add javadoc.

**Comment by bananaaggle** [comment]
> IMO, there should be an IT case to cover this change. But since there's no IT integration for community contributed extensions, currently we don't have to do so.
> 
> And for the changes in this PR, any javadoc to describe why `NoopQuerySegmentWalker` should be binded in router is useful.

Good suggestion! I will add javadoc.

**Comment by FrankChen021** [comment]
@bananaaggle My PR(#11262), in which a limitation description is added,  has been merged into the master branch, I'm suggesting you that you merge the master back to your branch and delete such limitation in this PR. Thanks.

**Comment by bananaaggle** [comment]
> @bananaaggle My PR(#11262), in which a limitation description is added, has been merged into the master branch, I'm suggesting you that you merge the master back to your branch and delete such limitation in this PR. Thanks.

OK, Thanks!

**Comment by bananaaggle** [comment]
Hi, @FrankChen021. I add annotation for NoopQuerySegmentWalker and move it to package org.apache.druid.server. It is an empty implementation, so should I create unit test for it? If necessary, where should I add it?

File: server/src/main/java/org/apache/druid/server/NoopQuerySegmentWalker.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import org.apache.druid.query.QuerySegmentWalker;
+import org.apache.druid.query.SegmentDescriptor;
+import org.joda.time.Interval;
+
+/**
+ * An empty implementation of {@link QuerySegmentWalker}.
+ *
+ * Some extentions need implementation of QuerySegmentWalker, but this class will not be used in
+ * router. Bind {@link NoopQuerySegmentWalker} in router to allow router load some extentions, which
+ * makes query can run
```
Comment (FrankChen021) [inline server/src/main/java/org/apache/druid/server/NoopQuerySegmentWalker.java:33]: nit: no new line after word 'run'

Comment (bananaaggle) [reply server/src/main/java/org/apache/druid/server/NoopQuerySegmentWalker.java]: Fixed.

**Review by FrankChen021 — APPROVED** [review]
LGTM 👍

**Review by clintropolis — APPROVED** [review]

**Pull request merged**